Assets/SEE/XR/Rotator.cs

```diff
@@ -4,185 +4,210 @@
+                            {
+                                return;
+                            }
+                            else
+                            {
+                                RotateDialClockwise();
+                                startAngle = currentAngle;
+                            }
                         }
-                        else
+                        else if (startAngle > currentAngle)
                         {
-                            RotateDialClockwise();
-                            startAngle = currentAngle;
+                            angleCheck = CheckAngle(currentAngle, startAngle);
+                            if (angleCheck < angleTolerance)
+                            {
+                                return;
+                            }
+                            else
+                            {
+                                RotateDialAntiClockwise();
+                                startAngle = currentAngle;
+                            }
                         }
                     }
-                    else if (startAngle > currentAngle)
+                    else
                     {
-                        angleCheck = CheckAngle(currentAngle, startAngle);
-                        if (angleCheck < angleTolerance)
+                        if (startAngle < currentAngle)
                         {
-                            return;
+                            RotateDialAntiClockwise();
                         }
-                        else
+                        else if (startAngle > currentAngle)
                         {
-                            RotateDialAntiClockwise();
-                            startAngle = currentAngle;
+                            RotateDialClockwise();
                         }
-                    }
-                }
-                else
-                {
-                    if (startAngle < currentAngle)
-                    {
-                        RotateDialAntiClockwise();
-                        startAngle = currentAngle;
-                    }
-                    else if (startAngle > currentAngle)
-                    {
-                        RotateDialClockwise();
                         startAngle = currentAngle;
                     }
                 }
             }
+            else
+            {
+                firstRotation = false;
+                startAngle = currentAngle;
+            }
         }
-        else
+
+        /// <summary>
+        /// Calculates the current rotation angle of the controller.
+        /// </summary>
+        /// <param name="currentAngle">The current angle of the controller.</param>
+        /// <param name="startAngle">The base angle.</param>
+        /// <returns>the amount of rotation</returns>
+        private float CheckAngle(float currentAngle, float startAngle) => (360f - currentAngle) + startAngle;
```

Comment:
A `Check` method name implies either a boolean return value, or a void return value with possible exceptions being thrown. 

However, I'm a bit confused by the documentation here, so I don't know what a good alternate name might be. Specifically, I'm confused by the documentation being "Calculates the current rotation angle of the controller" when one of the parameters is documented as "The current angle of the controller." Isn't this a bit contradictory?

Reply:
Well the current rotation angle of the controller could be 70 degrees, when the user starts the action, but the object that gets rotatet should not instantly rotate to that angle. Also we have a minimum angle which needs to be reached (angleTolerance) at which point the object gets rotated (otherwise it would be very hard to control for the user). So at the end we need to know, if the user really wanted to rotated the object or just had a shaky hand. For this purpose we calculate the amount of rotation, from the start point (e.g. 70 degrees) to the current point and use it afterwards to check if the rotation was big enough to actually cause a rotation of the object.

Reply:
But you are right, it is a bit confusing. I will change the code a bit and make it a bool, so it makes more sense.